electrum/address_synchronizer.py

```diff
@@ -987,6 +1000,7 @@ def get_utxos(
                     continue
                 coins.append(txo)
                 continue
+        self._get_utxos_cache[cache_key] = coins
```

Comment:
note that a deepcopy should be taken here as well, otherwise the first caller gets a chance to mutate the value in the cache.